### PR TITLE
feat: OB-35984 add conditional enabled values for each deployment

### PR DESCRIPTION
**Pull request by arthur-observe**
- add conditions to chart.yaml and values.yaml to provide override for each deployment

### CODE REVIEW

**Review by xueweiz — APPROVED** [review]

File: charts/agent/values.yaml [hunk 1]
```diff
@@ -36,6 +36,7 @@ config:
 
 ################################################
 deployment-cluster-events:
+  enabled: true
```
Comment (obs-gh-alexlew) [inline charts/agent/values.yaml:39]: i think these should go into the top level config vs. in here. also, im curious about the `config` top level field, isn't this all "config"? 

I think this field specifically should be `cluster.entities.enabled` or `cluster.resources.enabled`

File: charts/agent/values.yaml [hunk 2]
```diff
@@ -150,6 +151,7 @@ deployment-cluster-events:
 
 ################################################
 deployment-cluster-metrics:
+  enabled: true
```
Comment (obs-gh-alexlew) [inline charts/agent/values.yaml:154]: this field should be `cluster.metrics.enabled`

File: charts/agent/values.yaml [hunk 3]
```diff
@@ -262,6 +264,7 @@ deployment-cluster-metrics:
 
 ################################################
 daemonset-logs-metrics:
+  enabled: true
```
Comment (obs-gh-alexlew) [inline charts/agent/values.yaml:267]: really this should be split into 2 configs, `containers.logs.enabled` and `containers.metrics.enabled`

File: charts/agent/values.yaml [hunk 4]
```diff
@@ -406,6 +409,7 @@ daemonset-logs-metrics:
 
 ################################################
 deployment-agent-monitor:
+  enabled: true
```
Comment (obs-gh-alexlew) [inline charts/agent/values.yaml:412]: maybe this is just a top level config `agent_self_monitoring.enabled`

**Pull request merged**